src/document.rs

```diff
@@ -44,7 +44,7 @@ fn read_document<PB: Into<PathBuf>, P: AsRef<Path>>(root: PB, relpath: P) -> Res
     Ok(text)
 }
 
-fn split_document(content: &str) -> Result<(Option<&str>, &str)> {
+pub fn split_document(content: &str) -> Result<(Option<&str>, &str)> {
```

Comment:
You've made this and other code above `pub`.

Something to keep in mind when reusing code is if the reused code is inherent (same requirements) or incidental (same implementation but might diverge as requirements diverge).

I'd recommend creating a `split` designed for jekyll files.

(as for `read_file`, if its going to be reused, it should probably be moved to another `.rs` file).

Reply:
Ok I'll move `read_file` and see if a version of `split` for jekyll is a lot different or not :)